[NVPTX] Add 3-operand fmin/fmax DAGCombines

LewisCrawford opened the pull request:
Add DAGCombiner patterns for pairs of 2-operand min/max instructions to be fused into a single 3-operand min/max instruction for f32s (only for PTX 8.8+ and sm100+).

LewisCrawford committed:
[NVPTX] Add 3-operand fmin/fmax DAGCombines

Add DAGCombiner patterns for pairs of 2-operand min/max instructions
to be fused into a single 3-operand min/max instruction for f32s
(only for PTX 8.8+ and sm100+).

durga4github commented on llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp:
```diff
@@ -5316,6 +5320,56 @@ static SDValue PerformFADDCombine(SDNode *N,
+    SDValue B = Op0.getOperand(1);
+    SDValue C = Op1;
+    return DCI.DAG.getNode(MinMaxOp3, SDLoc(N), VT, A, B, C, N->getFlags());
+  } else if (Op1->getOpcode() == MinMaxOp2 && Op1->hasOneUse()) {
```
Just curious, why do we dereference using `->` here and using `.` in other places?

(Changes LGTM otherwise)

LewisCrawford replied:
There was no good reason, so I've changed it to use `.` in both places now.
Both `->` and `.` are equivalent in this case, as `*` returns the inner SDNode from an SDValue, and `getOperand` on an SDValue just calls `getOperand` on the inner SDNode anyway, but I've changed it to use the same approach everywhere now for consistency.

LewisCrawford committed:
Change Op1->getOperand to Op1.getOperand

Use . instead of -> operators on the SDValue Op1 for
consistency. Both are equivalent, as the * operator on
an SDValue returns the inner SDNode, and the getOperand
function on an SDValue just calls getOperand on the inner
node anyway, but we should use the same approach consistently.
